core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -511,23 +516,25 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
 
   private def processReconfiguration(newProps: Map[String, String], validateOnly: Boolean): (KafkaConfig, List[BrokerReconfigurable]) = {
     val newConfig = new KafkaConfig(newProps.asJava, !validateOnly, None)
-    val updatedMap = updatedConfigs(newConfig.originalsFromThisConfig, currentConfig.originals)
-    if (updatedMap.nonEmpty) {
+    val updatedConfig = updatedConfigs(newConfig.originalsFromThisConfig, currentConfig.originals)
```

Comment:
This works, but it looks a little nicer from a scala point of view to have `val (changeMap, deletedKeySet) = updatedConfigs(...)`.  It's a bit of an adjustment coming from Java, I know.

Reply:
Ah okay wasn't sure if I could do that - good to know. I have just made the change and pushed.